### ISSUE TITLE
Behavior changed when passing an array to `form_for` in Rails 7.0

### ISSUE THREAD

**Issue by krororo** [rails/rails#48789]
### Steps to reproduce

When passing an array of objects that can be call `to_model` to `form_for`, the `method: patch` is not included.
This issue is similar to #44326. However, it seems that nested routes have not been fixed.

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # gem "rails", github: "rails/rails", branch: "main"
  # gem "rails", "6.1.7.4"
  gem "rails", "7.0.6"
end

require "rack/test"
require "active_model"
require "action_view"
require "minitest/autorun"

class Post
  include ActiveModel::Model

  attr_reader :id

  def initialize(persisted: false)
    @persisted = persisted
    @id = 1 if persisted
  end

  def persisted?
    @persisted
  end
end

class PostForm
  include ActiveModel::Model

  def initialize(post)
    @post = post
  end

  def to_model
    @post
  end
end

class BugTest < ActionView::TestCase
  tests ActionView::Helpers::FormHelper

  routes = ActionDispatch::Routing::RouteSet.new
  routes.draw do
    namespace :admin do
      resources :posts
    end
  end
  include routes.url_helpers

  def test_form_for_namespaced_routes_new_model
    assert_dom_equal(
      '<form class="new_post" id="new_post" action="/admin/posts" accept-charset="UTF-8" method="post">' \
      '<input name="utf8" type="hidden" value="&#x2713;" autocomplete="off" />' \
      '</form>',
      form_for([:admin, PostForm.new(Post.new)]) {}
    )
  end

  def test_form_for_namespaced_routes_edit_model
    assert_dom_equal(
      '<form class="edit_post" id="edit_post_1" action="/admin/posts/1" accept-charset="UTF-8" method="post">' \
      '<input name="utf8" type="hidden" value="&#x2713;" autocomplete="off" />' \
      '<input name="_method" type="hidden" value="patch" autocomplete="off" />' \
      '</form>',
      form_for([:admin, PostForm.new(Post.new(persisted: true))]) {}
    )
  end
end
```

### Expected behavior

These tests are passed in Rails 6.1 and Rails main branch. So it should work on Rails 7.0 as well.

```
# Running:

..

Finished in 0.069481s, 28.7847 runs/s, 28.7847 assertions/s.
2 runs, 2 assertions, 0 failures, 0 errors, 0 skips
```

### Actual behavior

`test_form_for_namespaced_routes_edit_model` fails in Rails 7.0.
The `_method` of the hidden field is not included.

```
# Running:

F

Failure:
BugTest#test_form_for_namespaced_routes_edit_model [actionview_test.rb:68]:
Expected: <form class="edit_post" id="edit_post_1" action="/admin/posts/1" accept-charset="UTF-8" method="post"><input name="utf8" type="hidden" value="&#x2713;" autocomplete="off" /><input name="_method" type="hidden" value="patch" autocomplete="off" /></form>
Actual: <form class="edit_post" id="edit_post_1" action="/admin/posts/1" accept-charset="UTF-8" method="post"><input name="utf8" type="hidden" value="&#x2713;" autocomplete="off" /></form>


rails test actionview_test.rb:67

.

Finished in 0.036721s, 54.4642 runs/s, 54.4642 assertions/s.
2 runs, 2 assertions, 1 failures, 0 errors, 0 skips
```

### System configuration
**Rails version**:
7.0.6
**Ruby version**:
3.1.4

**Comment by ghiculescu** [comment]
It looks like https://github.com/rails/rails/pull/44468 is the PR that fixes this on `main`; that PR is not included in Rails 7.

**Comment by seanpdoyle** [comment]
> It looks like #44468 is the PR that fixes this on `main`; that PR is not included in Rails 7. Does that sound right to you @seanpdoyle ?

Right, I don't think there was a backport into 7.0

**Comment by ghiculescu** [comment]
I guess it's a bit confusing because the PR body describes it as a breaking change. But to be clear, you think it *should* be backported?

cc @byroot or @gmcgibbon if one of you guys agrees & is able to.

**Comment by byroot** [comment]
Hum, I haven't dug much yet, but at first glance I'm not sure this can be backported just like that.

**Comment by ghiculescu** [comment]
Okay, I guess I will close this as being fixed in 7.1 / edge Rails.
